### DISCUSSION TITLE
Add mass delete beatmaps

### DISCUSSION THREAD

**Discussion by TheLazyhog** [ppy/osu#31239] [Ideas]
i cant delete my old and easy beatmaps so i think this would be very useful, like when we choose diffuculty range there must be an option to delete all visible beatmaps

**Comment by syzyxy** [comment]
I agree, there's no Groups here unlike stable so can't do that right now. I'd also suggest maybe if a mode icon can be right-clickable, there could be a delete all maps option since sometimes people wanna delete all taiko maps for example... but yea a "delete all existing maps (in the current map selection screen)" like what stable has, would be nice.

**Comment by yeahirai** [comment]
Please i'm begging for something like this. Its such a pain to not be able to to this directly from lazer. What i had to do to be able to get the songs folder small enough to fit in my ssd was use BeatmapExporter to get the lazer maps back into .osz files. Then import all maps back to stable. Do the usual cleanup deleting small diffs, running osu cleaner to delete garbage files in the songs folders, then running another tool to delete folders without .osu files. Then importing everything back to lazer. 

Everytime i download a map it feels like i'm signing a contract where i'm either gonna have to keep this map forever, gonna need to manually delete 1 map at a time in the song select or to delete the entire songs folder, not part of the songs folder, but the whole thing. It's such a hassle. I wish is was taken more seriously considering how this storage system killed all 3rd party software that was used to deal with problems like these.

**Comment by mixalisnew** [comment]
i agree i have 80k beatmaps but the 5 star and above that i want are less than 20k so many of them are useless to me and i cant delete them unless i go grind mode

**Comment by SUKEsann2000** [comment]
I agree. I have a lot of beatmaps, and I want to remove part of them. But it's very hard for me.
I think removing them by our hands wastes a lot of time.

**Comment by SirHonkkk** [comment]
just add a “delete all beatmaps under ?⭐️” button. the “?” can be a selectable number, and you could lock some beatmaps to not get removed by this feature.

**Comment by Pufikas** [comment]
This would be awesome to have it under "Maintannce" tab, with difficulty slider and gamemode selections.
![Ca123123132pture](https://github.com/user-attachments/assets/d2c11c57-a206-429d-affc-b4e07873bbd7)

**Reply by SirHonkkk** [reply]
👍🏻

**Reply by n0Oo0Oo0b** [reply]
I think something like "Delete ALL visible maps" (only clickable in song select) makes more sense because it allows for more granularity and won't require re-implementing things like the SR slider in the options menu.

**Reply by I21b** [reply]
> I think something like "Delete ALL visible maps" (only clickable in song select) makes more sense

I agree, we can use the search filter like `status=g` to delete all filtered maps

**Comment by cerfeque** [comment]
Need option to delete all the maps that doesn't belong to any collection. Or delete groups
